Fixed crash on receiving request with "params": null

JonasRock opened the pull request:
When parsing correctly formed requests/notifications with
~~~~~~~~~~~~~~~~~~~
"params": null
~~~~~~~~~~~~~~~~~~~
a crash occurs. Checking if its null beforehand prevents this.

Commit:
Fixed crash on receiving request with "params": null

badaix commented:
thanks, can you please add a test case for this in [test_main.cpp](https://github.com/badaix/jsonrpcpp/blob/master/test/test_main.cpp)?

JonasRock committed:
fixed is_null for null parameters

JonasRock committed:
added test case for null parameters

JonasRock commented:
On writing the tests, I noticed that is_null() was not working for these parameters. Tests are helpful afterall, duh.
I fixed that and added the test case.

badaix commented:
great, thanks!
